### ISSUE TITLE
Add Predefined steps

### ISSUE THREAD

**Issue by mattapperson** [eridem/patata#18]
Just as calabash has generic predefined step definitions, I think patata should too:

## Screenshots
You can take a screenshot

- [ ] Then take a picture

## Touching (tapping)
Coordinate based touches
- [ ] Then I touch on screen 100 from the left and 250 from the top

Touching arbitrary views by accessibility label:
- [ ] Then I touch "accLabel"

Buttons by accessibility label
- [ ] Then I touch the "login" button

or by number/index (note number starts with 1 not 0)
- [ ] Then I touch button number 1

Input fields (UITextFields). Note, this looks for an UITextField with the placeholder property set to the quoted string.
- [ ] Then I touch the "placeholder" input field

List items (UITableViewCells) by number. Note this can only be used to touch visible cells - it doesn't try to scroll down/up.
- [ ] Then I touch list item number 1

Switches (UISwitch),
- [ ] Then I toggle the switch

this step works if there is a single switch. Otherwise use,
- [ ] Then I toggle the "accLabel" switch
where "accLabel" should be the accessibilityLabel for the switch.

Built-in keyboard. Touching the Done/Search button
- [ ] Then I touch done
- [ ] Then I touch search

## Map views.

- [ ] Then I touch the user location

## Entering text

Entering text by accessibility label:
- [ ] Then I enter "text to write" into the "accessibility label" input field
- [ ] Then I fill in "accessibility label" with "text to write"

Text by table (fields are by "placeholder")
- [ ] Then I fill in text fields as follows:
 | field      | text    |
 | Last Name  | Krukow  |
 | Email      | a@b.c   |
 | Username   | krukow  |
 | Password   | 123     |
 | Confirm    | 123     |

Text by input field number:
- [ ] Then I enter "text" into input field number 1

Clearing fields (like entering "")
- [ ] Then I clear "accessibility label"

Clearing fields by number
- [ ] Then I clear input field number 1

## Waiting

Waiting for text, or a view with a certain accessibilityLabel
- [ ] Then I wait to see "text or label"
- [ ] Then I wait for "text or label" to appear

wait for something to disappear
- [ ] Then I wait until I don't see "text or label"

waiting for a button with an accessibilityLabel
- [ ] Then I wait for the "login" button to appear

waiting for an iOS navigation bar with a certain title,
- [ ] Then I wait to see a navigation bar titled "title"

waiting for a text field
- [ ] Then I wait for the "label" input field

waiting for a number of text fields
- [ ] Then I wait for 2 input fields

waiting in general
- [ ] Then I wait
- [ ] Then I wait and wait
- [ ] Then I wait and wait and wait...
- [ ] Then I wait for 2.3 seconds

## Back button

In an iOS navigation bar, you can touch the "back" button using
- [ ] Then I go back

## Swipes

Swiping an unspecified place (usually when you have big scroll view in the center of the screen). Swipe directions can be left, right, up and down.

- [ ] Then I swipe left

Swiping a scroll view by index/number (and offset), or accessibilityLabel
- [ ] Then I swipe left on number 2
- [ ] Then I swipe left on number 2 at x 20 and y 10
- [ ] Then I swipe left on "accLabel"

Swiping table cells, by number
- [ ] Then I swipe on cell number 2

## Pinch (for zooms)

This step makes a small pinch to zoom in or out. Either at the first scroll view of the screen, or a the center of a view with a certain accessibility label.

- [ ] Then I pinch to zoom in
- [ ] Then I pinch to zoom in on "accLabel"

## Scrolling

Scrolling on scroll views. Direction can be left, right, up or down.
- [ ] Then I scroll down
- [ ] Then I scroll down on "accLabel"

## Device orientation

You can rotate the device or simulator left or right.
- [ ] Then I rotate device left

Assertions

Like waiting, these are usually about seeing certain text or ui components. Usually these are identified by their accessibilityLabels, component type (like a navigation bar) or pure text like in a label or a web view.

If the thing being asserted doesn't exist in the view, the test will fail.

Asserting existence of text, or a view with a certain accessibilityLabel

- [ ] Then I should see "text or label"
- [ ] Then I should not see "text or label"
- [ ] Then I see the text "some text"
- [ ] Then I don't see the text "text or label"
- [ ] Then I don't see the "someview"
- [ ] Then I see the "someview"

Asserting existence of buttons:
- [ ] Then I should see a "login" button
- [ ] Then I should not see a "login" button

More on text, prefix, suffix, and sub string.
- [ ] Then I should see text starting with "prefix"
- [ ] Then I should see text containing "sub text"
- [ ] Then I should see text ending with "suffix"

Seeing some text fields
- [ ] Then I see 2 input fields
- [ ] Then I should see a "Username" input field
- [ ] Then I should not see a "Username" input field

Seeing maps and user location
- [ ] Then I should see a map
- [ ] Then I should see the user location

**Comment by eridem** [comment]
Closed.
The QA developers should define their step definitions. Patata is a framework to speed up the implementation of the code.

**Comment by mattapperson** [comment]
Your assuming all QA persons can code, many can't. Including this would allow the framework to be instantly usable and devs only need to spend time adding project spacific features
